test/specs/dom/index.js

```diff
@@ -0,0 +1,545 @@
+    a.href = href;
+    div.createShadowRoot().appendChild(a);
+    parent.addEventListener("click", clickHandler);
+    fireEvent(a.firstChild, "click");
+
+    parent.removeEventListener("click", clickHandler);
+    parent.removeChild(div);
+  }
+  function findShadowDOMButton() {
+    const btn = document.querySelector("my-menu-button");
+    return btn && btn.innerText.indexOf("Shadow") > -1;
+  }
+  function createLinkTestNestedShadowDOMContent() {
+    const btn = document.querySelector("my-menu-button strong");
+    btn.click();
+  }
+
+  function clickHandler(event) {
+    event.preventDefault();
+  }
+  let instanceNumber = 0;
+
+  describe("Links", function() {
+    let currLoc;
+    before(function() {
+      currLoc = window.location.href;
+    });
+    after(function() {
+      history.pushState(null, null, currLoc);
+    });
+
+    describe("PalindromDOM - Links - ", function() {
+      describe("when attached to default node - `document.body`", function() {
+        let palindrom;
+        let historySpy;
+
+        beforeEach("PalindromDOM - Links", function(done) {
+          historySpy = sinon.spy(window.history, "pushState");
+
+          moxios.install();
+          moxios.stubRequest("http://localhost/testURL", {
+            status: 200,
+            headers: { location: "http://localhost/testURLNew" },
+            responseText: '{"hello": "world"}'
+          });
+
+          palindrom = new PalindromDOM({
+            remoteUrl: "http://localhost/testURL"
+          });
+          palindrom.instanceNumber = instanceNumber++;
+
+          setTimeout(done, 1);
+        });
+        afterEach(function() {
+          window.history.pushState.restore();
+          historySpy = null;
+          palindrom.unobserve();
+          palindrom.unlisten();
+          moxios.uninstall();
+        });
+
+        it("its `.element` should point to `document.body`", function() {
+          assert(palindrom.element === document.body);
+        });
+        describe("should intercept links to use History API", function() {
+          it("relative path", function() {
+            const href = "test_a";
+
+            createLinkTest(href);
+            expect(historySpy.callCount).to.equal(1);
+          });
+
+          it("relative path (nested)", function() {
+            const href = "test_b";
+
+            createLinkTestNested(href);
+            expect(historySpy.callCount).to.equal(1);
+          });
+
+          it("relative path (nested, Shadow DOM)", function() {
+            const href = "test_c";
+
+            try {
+              createLinkTestNestedShadowDOM(href);
+              expect(historySpy.callCount).to.equal(1);
+            } catch (e) {
+              console.warn(e);
+              expect(1).to.equal(1);
+            }
+          });
+          /* @tomalec would really appreciate some investigation on this,
+          the button ShadowDOM isn't being loaded sometimes (it doesn't become blue) */
```

Comment:
Turns out this have bitten you before, exactly the same issue https://github.com/webcomponents/webcomponentsjs/issues/575#issuecomment-258175058

![image](https://cloud.githubusercontent.com/assets/17054134/24799778/732aae30-1b9d-11e7-80cf-5545fd8614aa.png)